controlpanel/api/permissions.py

```diff
@@ -114,3 +114,26 @@ def has_object_permission(self, request, view, obj):
 class ToolDeploymentPermissions(BasePermission):
     def has_permission(self, request, view):
         return not request.user.is_anonymous
+
+
+class ParameterPermissions(BasePermission):
+    """
+    Superusers can do anything.
+    Normal users can list all, retrieve/update only themselves.
```

Comment:
I know it's an artifact of a Copy&Paste but probably worth changing this to

> Normal users can list all, retrieve/update only they parameters.

to avoid any possible confusion.

Reply:
Also, can all users see all Parameters really? If I remember correctly [the filtering](https://github.com/ministryofjustice/analytics-platform-control-panel/pull/705/files#diff-8bd403b7253dd1da943b0ca74d79d7d8R67) is done before the permissions are checked.

That should mean that users will only see the params they created (which sounds sensible for normal users). If that's the case this comment should really be:

> Normal users can list/retrieve/update only parameters they created.

(or something like that)

If that's the case it means that the UI don't need to do anything fancy to show/hide buttons to delete parameters as the list will only include parameters they can mess with (but worth double checking).

Reply:
This is why comments are rarely useful

Reply:
Also, I think if you are setting Parameters to be used as environment variables for a webapp, then the app admins should all be able to see and edit the list of parameters.

Reply:
Yeah, maybe something to do with Django Guardian